### PR TITLE
refactor and cleanup

### PR DESCRIPTION
**Pull request by duanyutong**
### Background

refactor and cleanup

### What's new
- ✅ refactor and cleanup
- ✅ publish franka exception details over lcm before and regardless of `RecoverFromControlException` outcome
- ✅ resolve a ton of linter warnings, including the include style
- ✅ updated hooks including license header
- ❌ New feature is accompanied by new test: [name and description of new test].

### Related work
- ❌ This PR needs [link]
- ❌ [link] needs this PR

### TODOs / Nice-To-Haves
- ❌ [Add new unit test for new feature.]
- ❌ [Add new system test for new feature.]

### Tests
1. ✅ Tested on simulated robot: [Describe the tests/commands used.]
2. ✅ Tested on physical robot: [Describe the tests/commands used.]

### Quality
3. ✅ New code is written in pure C++17 to the best of my knowledge.
4. ✅ New code follows established C++17 best practices ([C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines)).
5. ✅ New code passed `clang-format` and `cpplint` at least, if not all of our static code analysers.
6. ✅ I have included Doxygen-style documentation in the new C++ code.

### Note to reviewers
Please verify that all sections are accurately filled out and that all 6 numbered entries above are present and ticked/checked off, with their requirements met, if applicable.

### CODE REVIEW

File: src/driver/franka_plan_runner.cc [hunk 1]
```diff
@@ -171,159 +213,146 @@ franka::RobotMode FrankaPlanRunner::GetRobotMode(franka::Robot& robot) {
...
 int FrankaPlanRunner::RunFranka() {
-  bool connection_established {false};
-  // attempt connection to robot and read current mode
-  // and if it fails, keep trying instead of exiting the program
-  while (!connection_established) {
-    try {
-      franka::Robot robot(ip_addr_);
-
-      auto current_mode {GetRobotMode(robot)};
+  {  // connection and mode checking and handling
```
Comment (duanyutong) [inline src/driver/franka_plan_runner.cc:216]: main loop refactor starts here

File: src/driver/franka_plan_runner.cc [hunk 2]
```diff
@@ -171,159 +213,146 @@ franka::RobotMode FrankaPlanRunner::GetRobotMode(franka::Robot& robot) {
...
+        dexai::log()->warn("RunFranka: control exception in main loop: {}.",
                            ce.what());
-
-        if (!RecoverFromControlException(robot)) {
-          dexai::log()->error(
-              "RunFranka: RecoverFromControlException did not work!");
-          comm_interface_->PublishDriverStatus(false, ce.what());
+        comm_interface_->PublishDriverStatus(false, ce.what());
+        if (!RecoverFromControlException(*robot_)) {
+          dexai::log()->error("RunFranka: RecoverFromControlException failed");
```
Comment (duanyutong) [inline src/driver/franka_plan_runner.cc:308]: publish franka exception details over lcm before and regardless of `RecoverFromControlException` outcome

File: src/utils/util_conv.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+///@file util_conv.h
+#pragma once
+
+#include "Eigen/Geometry"
+#include "drake/lcmt_iiwa_status.hpp"  // for lcmt_iiwa_status
+#include "driver/communication_interface.h"
+#include "franka/robot_state.h"  // for RobotState, RobotMode
+
+/// NOTE @sprax: I doubt there is still a need to organized these functions
+/// here. They could be elsewhere or be eliminated.
```
Comment (dmsj) [inline src/utils/util_conv.h:43]: remove note

File: src/driver/franka_plan_runner.cc [hunk 4]
```diff
@@ -171,159 +213,146 @@ franka::RobotMode FrankaPlanRunner::GetRobotMode(franka::Robot& robot) {
...
+    if (comm_interface_->CancelPlanRequested()) {
+      log()->error(
+          "RunFranka: plan cancellation requested but there is no active "
+          "plan");
+      comm_interface_->ClearCancelPlanRequest();
+    }
+    // print out status if there's any change, or every 10 sec
+    if (auto t_now {std::chrono::steady_clock::now()};
+        status_has_changed
+        || t_now - t_last_main_loop_log_ >= std::chrono::seconds(10)) {
```
Comment (dmsj) [inline src/driver/franka_plan_runner.cc:324]: How should we specify these wait times? We have a 1-second wait earlier and a 10-second wait here.

**Review by dmsj — COMMENTED** [review]
looks pretty good! needs on-robot testing for full sign-off

File: src/driver/communication_interface.cc [hunk 5]
```diff
@@ -12,24 +45,21 @@
 /// If a pause message is received, it will set the opause status to true and
```
Comment (5yler) [inline src/driver/communication_interface.cc:45]: "opause status"?

File: src/driver/communication_interface.h [hunk 6]
```diff
@@ -73,7 +108,8 @@ class CommunicationInterface {
 
   std::tuple<std::unique_ptr<PPType>, int64_t> PopNewPlan();
 
-  // TODO: remove franka specific RobotState type and replace with std::array
+  // TODO(@anyone): remove franka specific RobotState type and
+  // replace with std::array
   franka::RobotState GetRobotState();
```
Comment (5yler) [inline src/driver/communication_interface.h:113]: why?

File: src/driver/franka_plan_runner.cc [hunk 7]
```diff
@@ -171,159 +213,146 @@ franka::RobotMode FrankaPlanRunner::GetRobotMode(franka::Robot& robot) {
...
+    bool connection_established {};
+    do {  // do-while(!connection_established) loop, execute once first
+      try {
+        robot_ = std::make_unique<franka::Robot>(ip_addr_);
+      } catch (franka::Exception const& e) {
+        // probably something wrong with networking
+        auto err_msg {fmt::format("Franka connection error: {}", e.what())};
+        dexai::log()->error("RunFranka: {}", err_msg);
+        comm_interface_->PublishDriverStatus(false, err_msg);
+        std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```
[the remark this answers is not in the document]
Comment (5yler) [reply src/driver/franka_plan_runner.cc:228]: If there's a networking problem, there's no point in retrying more frequently. If we hit this code block something is very wrong and the controller is likely not connected or not communicating

File: src/driver/franka_plan_runner.cc (hunk 2 above)
Comment (5yler) [reply src/driver/franka_plan_runner.cc:308]: The driver status LCM message with `success = false` indicates that the driver is not running. A successful recovery from control exception shouldn't have a success of `false`

**Review by 5yler — COMMENTED** [review]
:heart_eyes_cat:

File: src/driver/franka_plan_runner.cc (hunk 2 above)
Comment (duanyutong) [reply src/driver/franka_plan_runner.cc]: updated.

File: src/utils/util_conv.h (hunk 3 above)
Comment (duanyutong) [reply src/utils/util_conv.h:43]: removed

File: src/driver/franka_plan_runner.cc (hunk 4 above)
Comment (duanyutong) [reply src/driver/franka_plan_runner.cc:271]: they are different things, this is only for the status printout

File: src/driver/communication_interface.h (hunk 6 above)
Comment (duanyutong) [reply src/driver/communication_interface.h:113]: typo? changed to pause

File: src/driver/communication_interface.cc (hunk 5 above)
Comment (duanyutong) [reply src/driver/communication_interface.cc:45]: typo? changed to pause

**Review by 5yler — APPROVED** [review]
LGTM platform & style assuming sufficiently tested on robot

**Pull request merged**